### PR TITLE
Extract service implementation from go-libp2p-autonat

### PR DESCRIPTION
**Pull request by vyzo**
Depends on:
- [X] https://github.com/libp2p/go-libp2p-autonat/pull/2
- [X] https://github.com/multiformats/go-multiaddr-net/pull/46

TBD:
- [X] gx package
- [X] update go-libp2p dep once changes have bubbled up

### CODE REVIEW

**Comment by vyzo** [comment]
we will need it for deployment, but it's not in the critical path yet.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# go-libp2p-discovery
```
Comment (raulk) [inline README.md]:
```suggestion
# go-libp2p-autonat-svc
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+[![](https://img.shields.io/badge/made%20by-Protocol%20Labs-blue.svg?style=flat-square)](http://ipn.io)
+[![](https://img.shields.io/badge/project-libp2p-blue.svg?style=flat-square)](http://libp2p.io/)
+[![](https://img.shields.io/badge/freenode-%23ipfs-blue.svg?style=flat-square)](http://webchat.freenode.net/?channels=%23ipfs)
+[![standard-readme compliant](https://img.shields.io/badge/standard--readme-OK-green.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)
+
+> AutoNAT service implementation
+
+This package provides an implementation of the AutoNATService; see [autonat](https://github.com/libp2p/go-libp2p-autonat).
+
+## Documenation
```
Comment (raulk) [inline README.md]:
```suggestion
## Documentation
```

**Review by raulk — COMMENTED** [review]
LGTM. Mostly shuffling around code that has already been reviewed. Just some typos on the docs.

**Review by raulk — APPROVED** [review]

**Comment by vyzo** [comment]
We are blocked on merging, waiting for the changes in go-multiaddr-net to bubble up.

**Pull request merged**